openmetadata-service/src/main/java/org/openmetadata/service/apps/AbstractNativeApplication.java

```diff
@@ -205,6 +210,20 @@ private void createAndBindIngestionPipeline(
             .getByName(null, SERVICE_NAME, serviceEntityRepository.getFields("id"))
             .getEntityReference();
 
+    // Get encrypted private config
+    Object privateConfig = this.getApp().getPrivateConfiguration();
+    if (privateConfig != null) {
+      String privateConfigJson = JsonUtils.pojoToJson(privateConfig);
+      SecretsManager secretsManager = SecretsManagerFactory.getSecretsManager();
+      if (secretsManager instanceof ExternalSecretsManager) {
+        String baseSecretId =
+            "external-app-" + this.getApp().getName().toLowerCase() + "-private-config";
+        privateConfig = "secret:" + baseSecretId;
```

Comment:
should we have such important keys in here? `secret:` is a constant already in the SecretsManager class. I feel a bit uneasy with the `external-app-` prefix handling directly here. Not sure if we could centralize this logic on the SecretsMaanger itself vs. having it scattered